modules/core/ui/views/tests/src/Functional/FarmUiViewsTest.php

```diff
@@ -21,20 +22,49 @@ class FarmUiViewsTest extends FarmBrowserTestBase {
   protected static $modules = [
     'farm_activity',
     'farm_equipment',
+    'farm_inventory',
+    'farm_observation',
+    'farm_quantity_standard',
     'farm_water',
     'farm_ui_views',
     'farm_ui_views_test',
   ];
 
   /**
-   * Test Views provided by the farm_ui_views module.
+   * {@inheritdoc}
    */
-  public function testFarmUiViews() {
+  protected function setUp(): void {
+    parent::setUp();
 
-    // Create and login a user with permission to view assets.
-    $user = $this->createUser(['view any asset', 'view any log']);
+    // Create and login a user with permission to view assets, logs, and
+    // quantities.
+    $user = $this->createUser(['view any asset', 'view any log', 'view any quantity']);
     $this->drupalLogin($user);
 
+    // Disable entity_reference_integrity_enforce module's protections, so we
+    // can delete all entities easily.
+    $erie_config = \Drupal::configFactory()->getEditable('entity_reference_integrity_enforce.settings');
+    $erie_config->set('enabled_entity_type_ids', []);
+    $erie_config->save();
+  }
+
+  /**
+   * Run all tests.
+   */
+  public function testAll() {
+    $this->doTestAssetViews();
+    $this->doTestLogViews();
+    $this->doTestAssetsByLocationView();
+    $this->doTestAssetChildrenView();
+    $this->doTestAssetInventoryView();
+    $this->doTestAssetLogsView();
+  }
```

Comment:
Is there a reason for combining here instead of running in separate test methods? Is this order important or are there dependencies between test methods?

It seems this introduces the need for `deleteAllEntities()` method and other complexity?

Reply:
This is the reason: https://github.com/farmOS/farmOS/pull/797

Combining functional tests reduces test run time.

> Is this order important or are there dependencies between test methods?

No the order doesn't matter, and yes I added the `deleteAllEntities()` logic to keep everything clean and separate between them. I think that the tests would still pass if we removed the `deleteAllEntities()` - but it felt better to have it explicit, to prevent potential issues in the future.

Splitting them up into separate tests would achieve the same goal, at the expense of longer test run time. I felt that these were all similar enough that it didn't warrant separate bootstraps, but didn't want to have them all in the same test method that we started with (https://github.com/farmOS/farmOS/pull/952/commits/c81ed75f67b70fb2a436e9bc30082a07c19e0d96), so I split them into separate methods run by `testAll()`, and added the `deleteAllEntities()` logic as a simple way to clean up in between. The same `deleteAllEntities()` code could be used if we kept them all in the same method too, so it feels natural to keep it IMO.

Reply:
Makes sense. If the tests were all very different I'd argue to keep them separate but agree, all the logic is quite similar here.

Maybe moving `deleteAllEntities()` into the `testAll()` function would make this dynamic easier to understand? Normally test methods don't need to do any cleanup.

```suggestion
  /**
   * Run all tests.
   */
  public function testAll() {
  
    // Run each farm UI views test in a single method to decrease test time.
    // Delete all entities between running each test.
    $this->doTestAssetViews();
    $this->deleteAllEntities();
    
    $this->doTestLogViews();
    $this->deleteAllEntities();
    
    $this->doTestAssetsByLocationView();
    $this->deleteAllEntities();
    
    $this->doTestAssetChildrenView();
    $this->deleteAllEntities();
    
    $this->doTestAssetInventoryView();
    $this->deleteAllEntities();
    
    $this->doTestAssetLogsView();
    $this->deleteAllEntities();
  }
```

Reply:
Ah yea I like that idea! Standby for a fixup...

Reply:
https://github.com/mstenta/farmOS/commit/b44dc9d4e5940152780bafab81d9d829b50a69df